app/Http/Controllers/Api/CityProviderController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\CityProviderRequest;
+use App\Http\Resources\CityResource;
+use App\Http\Resources\CountryResource;
+use App\Http\Resources\ProviderResource;
+use App\Models\City;
+use App\Models\Country;
+use App\Models\Provider;
+use App\Services\CityProviderService;
+use App\Services\DataImporterService;
+use Illuminate\Http\JsonResponse;
+
+class CityProviderController extends Controller
+{
+    public function index(): JsonResponse
+    {
+        $cities = CityResource::collection(
+            City::with("cityAlternativeNames", "cityProviders", "country")
+                ->has("cityProviders")
+                ->whereHas("cityProviders", function ($query): void {
+                    $query->whereNotNull("latitude")->whereNotNull("longitude");
```

Comment:
```suggestion
                ->whereHas("cityProviders", function ($query): Builder {
                    $query->whereNotNull("latitude")->whereNotNull("longitude");
```

It should be `builder` return type or smth like that.